src/routes/auth/providers/spotify.ts

```diff
@@ -0,0 +1,19 @@
+import { Router } from 'express'
+import { Strategy } from 'passport-spotify'
+import Boom from '@hapi/boom'
+import { PROVIDERS } from '@shared/config'
+import { initProvider } from './utils'
+
+export default (router: Router): void => {
+  const options = PROVIDERS.spotify
+  // Checks if the strategy is enabled. Don't create any route otherwise
+  if (options) {
+    // Checks if the strategy has at least a client ID and a client secret
+    if (!options.clientID || !options.clientSecret) {
+      throw Boom.badImplementation(`Missing environment variables for Spotify OAuth.`)
+    }
+    initProvider(router, 'spotify', Strategy, {
+        scope: ['user-read-email', 'user-read-private', 'app-remote-control', 'streaming']
```

Comment:
I don't think wee need all these scopes by default. I think we can use `'user-read-email', 'user-read-private'` only.

If the app needs more access I think we should do that in a separate PR with: 

- scope per provider
- save access tokens to providers

Reply:
@elitan how about making provider scopes configurable via env vars? I could make separate PR for that.

Reply:
> @elitan how about making provider scopes configurable via env vars? I could make separate PR for that.

Ye. I think it make sense. The scope should be configurable for each provider.

Maybe you could open a new issue about it and we could take the discussion there? Some initial thoughts:

- With extra permission I think we want to store the OAuth token (https://github.com/nhost/hasura-backend-plus/pull/367). So that you could ex use the OAuth token to change song on Spotify?
- Do some provider rotate the OAuth token?